### PR TITLE
Upgrade all projects to .NET6.0

### PR DESCRIPTION
**Pull request by yopichy**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/elsa-workflows/elsa-core?pullRequest=2616) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/elsa-workflows/elsa-core?pullRequest=2616) before we can accept your contribution.<br/><br/>:x: yopichy [sign now](https://cla.dotnetfoundation.org/elsa-workflows/elsa-core?pullRequest=2616)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/elsa-workflows/elsa-core?pullRequest=2616) it.</sub>

**Comment by yopichy** [comment]
@sfmskywalker CI .NET SDK does not support targeting .NET 6.0

**Comment by sfmskywalker** [comment]
@yopichy Thank you for this! I'll update the CI settings.

**Comment by sfmskywalker** [comment]
@yopichy Question about this PR: will .NET 5 project still be able to use the Elsa core libraries, Entity Framework etc?

**Comment by yopichy** [comment]
@sfmskywalker 
Elsa core works with NET 5, but not with EF.
If i removing EFCore.BulkExtensions then it can run on NET 5, but for now I keep the bulk command feature in this PR.
But if you want compatibility with NET 5 I can make it, but sacrifice the bulk command feature or have to replace EFCore.BulkExtensions

**Comment by sfmskywalker** [comment]
@yopichy We should definitely keep support for .NET 5 for the time being. We should be able to conditionally target both versions of e.g. EFCore.BulkExtensions, right?

E.g.

```xml
<ItemGroup Condition=" '$(TargetFramework)' == 'net6.0'">
    <PackageReference Include="EFCore.BulkExtensions" Version="6.x" />
</ItemGroup>

<ItemGroup Condition=" '$(TargetFramework)' == 'net5.0'">
    <PackageReference Include="EFCore.BulkExtensions" Version="5.x" />
</ItemGroup>

<ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1'">
    <PackageReference Include="EFCore.BulkExtensions" Version="3.x" />
</ItemGroup>
```

**Comment by yopichy** [comment]
@sfmskywalker ok i'll try it

**Comment by sfmskywalker** [comment]
@yopichy Thank you 🙏🏻

**Comment by yopichy** [comment]
> @yopichy Thank you 🙏🏻

Done

File: src/activities/Elsa.Activities.Http/Elsa.Activities.Http.csproj [hunk 1]
```diff
@@ -15,26 +15,19 @@
...
         <PackageTags>elsa, workflows, http</PackageTags>
     </PropertyGroup>
-
-    <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0'">
-        <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="5.0.10" />
-    </ItemGroup>
-
-    <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1'">
-        <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="3.1.19" />
-    </ItemGroup>
```
Comment (sfmskywalker) [inline src/activities/Elsa.Activities.Http/Elsa.Activities.Http.csproj:25]: Will the removal of these references not break applications targeting .NET 5 and .NET 3.1?

**Review by mohdali — CHANGES_REQUESTED** [review]
Some changes are required to maintain 3.1 support

File: test/component/Elsa.ComponentTests/Elsa.ComponentTests.csproj [hunk 2]
```diff
@@ -1,18 +1,25 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net5.0</TargetFramework>
+        <TargetFrameworks>net5.0;net6.0</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply test/component/Elsa.ComponentTests/Elsa.ComponentTests.csproj:4]: Yes, samples & tests should target latest framework only. Only nuget packages need multi-targeting.

**Review by mohdali — CHANGES_REQUESTED** [review]
minimal changes to get monolith to start running on 3.1, additional changes still required for other persistence packages

**Comment by yopichy** [comment]
@mohdali @sfmskywalker i'll updated according to suggestion ASAP

File: src/server/Elsa.Server.Api/Elsa.Server.Api.csproj [hunk 3]
```diff
@@ -13,6 +13,18 @@
...
+    
+    <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0'">
+        <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="5.0.10" />
+    </ItemGroup>
+
+    <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0'">
+        <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.1" />
+    </ItemGroup>
+
     <ItemGroup>
```
Comment (mohdali) [inline src/server/Elsa.Server.Api/Elsa.Server.Api.csproj:28]:
```suggestion
    <ItemGroup>
            <FrameworkReference Include="Microsoft.AspNetCore.App" />
```

**Comment by mohdali** [comment]
@yopichy thank you very much. I went ahead and pushed some updates to implement some of the changes. hope it helps.

**Comment by yopichy** [comment]
Thanks for the help @mohdali

**Review by sfmskywalker — APPROVED** [review]

**Comment by DoomerDGR8** [comment]
This major change affect the Elsa 2.x only?

**Comment by sfmskywalker** [comment]
More specifically, it only affects `2.4.1` and up.

**Pull request merged**